src/pull_request.rs

```diff
@@ -310,52 +274,65 @@ pub fn handle_pull_request(
 		v.sort_by_key(|s| s.updated_at);
 		v.last()
 	});
-	if let Some(ref status) = status {
-		if status.state.as_deref() == Some("failure") {
-			// notify PR author by PM every 24 hours
-			let should_ping =
-				db_entry.status_failure_ping.map_or(true, |ping_time| {
-					ping_time.elapsed().ok().map_or(true, |elapsed| {
-						elapsed.as_secs() > STATUS_FAILURE_PING_PERIOD
-					})
-				});
 
-			if should_ping {
-				db_entry.status_failure_ping = Some(SystemTime::now());
-				if let Some(matrix_id) = github_to_matrix
-					.get(&repo.owner.login)
-					.and_then(|matrix_id| matrix::parse_id(matrix_id))
-				{
-					matrix_bot.send_private_message(
-						&matrix_id,
-						&STATUS_FAILURE_NOTIFICATION
-							.replace("{1}", &pr_html_url),
-					)?;
-				} else {
-					log::warn!("Couldn't send a message to {}; either their Github or Matrix handle is not set in Bamboo", &repo.owner.login);
+	match if let Some(ref status) = status {
```

Comment:
I would definitely break this up. A `match` within an `if` within a `match` is definitely not good style. 😄